software/control/microcontroller.py

```diff
@@ -339,6 +347,37 @@ def move_theta_usteps(self,usteps):
         # while self.mcu_cmd_execution_in_progress == True:
         #     time.sleep(self._motion_status_checking_interval)
 
+    def move_w_usteps(self,usteps):
+        direction = STAGE_MOVEMENT_SIGN_W*np.sign(usteps)
+        n_microsteps_abs = abs(usteps)
+        # if n_microsteps_abs exceed the max value that can be sent in one go
+        while n_microsteps_abs >= (2**32)/2:
+            n_microsteps_partial_abs = (2**32)/2 - 1
+            n_microsteps_partial = direction*n_microsteps_partial_abs
+            payload = self._int_to_payload(n_microsteps_partial,4)
+            cmd = bytearray(self.tx_buffer_length)
+            cmd[1] = CMD_SET.MOVE_W
+            cmd[2] = payload >> 24
+            cmd[3] = (payload >> 16) & 0xff
+            cmd[4] = (payload >> 8) & 0xff
+            cmd[5] = payload & 0xff
+            self.send_command(cmd)
+            # while self.mcu_cmd_execution_in_progress == True:
```

Comment:
You can remove all these commented out `while` (here and a bunch below).

Do they need a `wait_till_operation_is_complete()` instead?

Reply:
In some cases, move axis need to wait the movement finish, but the other cases don't need wait. So we move the wait operation out of the move function to a higher level logic code. I would remove the comments instead.